src/webui/static/app/tvheadend.js

```diff
@@ -1387,38 +1387,48 @@ tvheadend.toLocaleFormat = function()
 	return tvh_locale_lang.replace('_','-');
 };
 
-tvheadend.toCustomDate = function(date, format) //author: meizz, improvements by pablozg
-{
-    if(/([%][MmsSyYdhq]+)/.test(format)){
-        var o = {
-            "\%M+" : date.getMonth()+1, //month
-            "\%d+" : date.getDate(),    //day
-            "\%h+" : date.getHours(),   //hour
-            "\%m+" : date.getMinutes(), //minute
-            "\%s+" : date.getSeconds(), //second
-            "\%q+" : Math.floor((date.getMonth()+3)/3),  //quarter
-            "\%S" : date.getMilliseconds() //millisecond
-        }
-
-        if(/(\%[yY]+)/.test(format)) format=format.replace(RegExp.$1, (date.getFullYear()+"").substr(5 - RegExp.$1.length));
-
-        if(/(\%MMMM)/.test(format)) format=format.replace(RegExp.$1, (date.toLocaleDateString(tvheadend.toLocaleFormat(), {month: 'long'})));
-
-        if(/(\%MMM)/.test(format)) format=format.replace(RegExp.$1, (date.toLocaleDateString(tvheadend.toLocaleFormat(), {month: 'short'})));
+tvheadend.toCustomDate = function(date, format) {
+    if (/([%][MmsSyYdhHIpPq]+)/.test(format)) {
```

Comment:
Replace the `[%]` with `%`, no need for a character class when there is only a single character.